### PR TITLE
add naive approach for resizing the images

### PR DESCRIPTION
**Pull request by EnnioEnnio**
Currently Images are stretched to fit 256x256 causing change in aspect ratio.
somehow rotation occurs sometimes which led me to the idea that our CNN should be able to detect that as well - what do you think?  

I read [here on cs.se](https://cs.stackexchange.com/questions/134628/how-does-cnn-deal-with-rotation-invariant-pictures) that a simple method would be to just rotate training images randomly and train for more epochs.

### CODE REVIEW

File: data_loader.py [hunk 1]
```diff
@@ -9,3 +9,25 @@
 Obserservations:
 Images from the dataset are of different sizes.
 """
+
+import os
+from PIL import Image
```
Comment (Toalaah) [inline data_loader.py:14]: This is not part of the stdlib, no? I would add this to `requirements.txt`

**Review by Toalaah — CHANGES_REQUESTED** [review]
Apologies for the delayed response, was busy that last couple of days. I think that rotation detection is definitely a nice feature to support!

WRT normalization of the image sizes, the only other thing that came to mind for me is possibly using 0-padding instead of resizing. That way features are not stretched / compressed, but TBH I am not entirely sure on whether this is even that much of an issue for the CNN. I think this is a fine approach for now.

Furthermore, I suppose that padding would require more training epochs as theree are more possible locations fot the actual image to be placed inside of the padded canvas (actually, a quick search shows results which suggest the [opposite](https://journalofbigdata.springeropen.com/articles/10.1186/s40537-019-0263-7) in certain conditions). Perhaps we could add processing options for both in the future, giving us something to compare against for the report? Would be happy to implement this in a future PR.

File: data_loader.py (hunk 1 above)
Comment (EnnioEnnio) [reply data_loader.py:14]: Ah funny, didn't know that - import just worked without adding it to my conda env.
Added it to `requirements.txt`, thanks for the hint!

**Comment by EnnioEnnio** [comment]
Absolutely no worries, I also try to keep my weekends free from Uni :) 
Re rotation detection: Do you want rotation to be detected on input? I originally meant that it would be nice if it didn't matter for the recognition if the image is rotated or not. 


Re zero-padding: I initially wanted to simply add padding so as not to distort the images, but didn't have enough time to do so, hence "naive approach" in the title. In the next iteration then 😅. 
We could also try both and see if the performance improves, or degrades.

Regarding the position of the image I feel that this might not matter as they are sensitive to spacial relations but not the specific location of a feature (translational invariance) - but I might be wrong here.

**Comment by Toalaah** [comment]
> I originally meant that it would be nice if it didn't matter for the recognition if the image is rotated or not.

This is also what I understood regarding rotation. Let us do it like this.
 
> We could also try both and see if the performance improves, or degrades.

Sounds good :+1:

**Pull request merged**